tests/functional/test_changing_partitions.py

```diff
@@ -1,34 +1,88 @@
-
-    @property
-    def schema(self):
-        return "bigquery_test"
-
-    @property
+from dbt.tests.util import run_dbt
+
+_MACRO_SQL = """
+{% test number_partitions(model, expected) %}
+
+    {%- set result = get_partitions_metadata(model) %}
+
+    {% if result %}
+        {% set partitions = result.columns['partition_id'].values() %}
+    {% else %}
+        {% set partitions = () %}
+    {% endif %}
+
+    {% set actual = partitions | length %}
+    {% set success = 1 if model and actual == expected else 0 %}
+
+    select 'Expected {{ expected }}, but got {{ actual }}' as validation_error
+    from (select true)
+    where {{ success }} = 0
+
+{% endtest %}
+"""
+
+_MODEL_SQL = """
+{{
+    config(
+        materialized="table",
+        partition_by=var('partition_by'),
+        cluster_by=var('cluster_by'),
+        partition_expiration_days=var('partition_expiration_days'),
+        require_partition_filter=var('require_partition_filter')
+    )
+}}
+
+select 1 as id, 'dr. bigquery' as name, current_timestamp() as cur_time, current_date() as cur_date
+union all
+select 2 as id, 'prof. bigquery' as name, current_timestamp() as cur_time, current_date() as cur_date
+"""
+
+_SCHEMA_YML = """
+version: 2
+models:
+- name: my_model
+  tests:
+  - number_partitions:
+      expected: "{{ var('expected', 1) }}"
+"""
+
+
+class BaseBigQueryChangingPartition:
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {
+            "partition_metadata.sql": _MACRO_SQL
+        }
+
+    @pytest.fixture(scope='class')
     def models(self):
-        return "partition-models"
+        return {
+            "my_model.sql": _MODEL_SQL,
+            "schema.yml": _SCHEMA_YML
+        }
+
 
     def run_changes(self, before, after):
-        results = self.run_dbt(['run', '--vars', json.dumps(before)])
-        self.assertEqual(len(results), 1)
+        results = run_dbt(['run', '--vars', json.dumps(before)])
+        assert len(results) == 1
 
-        results = self.run_dbt(['run', '--vars', json.dumps(after)])
-        self.assertEqual(len(results), 1)
+        results = run_dbt(['run', '--vars', json.dumps(after)])
+        assert len(results) == 1
 
-    def test_partitions(self, expected):
-        test_results = self.run_dbt(['test', '--vars', json.dumps(expected)])
+    def partitions_test(self, expected):
+        test_results = run_dbt(['test', '--vars', json.dumps(expected)])
 
         for result in test_results:
-            self.assertEqual(result.status, 'pass')
-            self.assertFalse(result.skipped)
-            self.assertEqual(result.failures, 0)
+            assert result.status == "pass"
+            assert result.skipped == False
+            assert result.failures == 0
+
+class TestBigQueryChangingPartition(BaseBigQueryChangingPartition):
 
-    @use_profile('bigquery')
-    def test_bigquery_add_partition(self):
+    def test_bigquery_add_partition(self, project):
```

Comment:
nit: if you're mucking around these files, might as well remove `bigquery` from all these functions and stuff. I've been nit-ed on this in dbt-core for leaving postgres in places so might as well keep up with the idiom there 🤷‍♀️